### PR TITLE
DF-548:Integration for Archives details page

### PR DESCRIPTION
**Pull request by TNA-Allan**
Ticket URL: https://national-archives.atlassian.net/browse/DF-548

## About these changes

Add new archive details page

## How to check these changes

- Navigate to Catalogue Search - Find an archive bucket,
- select the archive record from the search results

## Before assigning to reviewer, please make sure you have

- [x] Checked things thoroughly before handing over to reviewer.
- [x] Checked PR title starts with ticket number as per project conventions to help us keep track of changes.
- [x] Ensured that PR includes only commits relevant to the ticket.
- [x] Waited for all CI jobs to pass before requesting a review.
- [x] Added/updated tests and documentation where relevant.

## Merging PR guidance

Follow [docs\developer-guide\contributing.md](https://nationalarchives.github.io/ds-wagtail/developer-guide/contributing/)

## Deployment guidance

Follow [docs\infra\environments.md](https://nationalarchives.github.io/ds-wagtail/infra/environments/)

### CODE REVIEW

File: etna/records/models.py
Comment (ababic) [inline etna/records/models.py:605]: Is there any reason why these two data structures need to be collected into a single object/value in this way? Might it be simpler to just access each value separately in the template?

**Review by ababic — CHANGES_REQUESTED** [review]
Thanks @TNA-Allan. There's quite a bit going on here. First of all, would it be possible to split the following out into separate PRs?
- Changing `title` to `summary_title`
- Changing `catalogue_source` to `source`
- Front-end changes (scss and template contents)

Separately, I would say that these new `Record` methods definitely require unit tests. Maybe with so much going on in each method, it might actually be good to move the "XML to dataclass instance conversion" logic to the dataclasses themselves (e.g. by adding a `from_xml()` classmethod)

File: etna/ciim/tests/factories.py
Comment (TNA-Allan) [reply etna/ciim/tests/factories.py]: Yes, its a typo. Its fixed to say `source`.

File: etna/records/models.py
Comment (TNA-Allan) [reply etna/records/models.py:605]: The Record Creators and NRA records are from different sources but are the under  Collections - in terms according to the [api ticket](https://national-archives.atlassian.net/browse/DF-537) . Also NRA records paper_catalogue, appears under Summary in the static html prototype. Its fields output is "somewhat similar" to NRA in terms of presentation. So due to these similarities collected into one object.

File: etna/ciim/constants.py
Comment (TNA-Allan) [reply etna/ciim/constants.py:721]: Nice one, and I agree with the puspose of another file, perhaps `data_classes.py` to be more specific. I also think it would fine if were to do this at that same later point, since also we have the following [data classes in constants.py](https://github.com/nationalarchives/ds-wagtail/blob/develop/etna/ciim/constants.py#L31-L67) dataclass-
`class Bucket:, class BucketList: `

**Comment by TNA-Allan** [comment]
> * Changing `title` to `summary_title`

We've never used the exclusive `title` attribute from the api response, until now in this PR. 

The [changes from `title` to `summary_title`](https://github.com/nationalarchives/ds-wagtail/pull/623/files#diff-dcce4f0507b1263688175bf965e5ac8d20492f9888c57a2a0a7f4c74dba00d55R166) were corrected over here to reflect what the API returned and used in the code base, the earlier `def title` was returning `summaryTitle`

There is also a new test case for this PR [def test_title:](https://github.com/nationalarchives/ds-wagtail/blob/6c795406ff11836b456da72252c411fa48d05a8e/etna/records/tests/test_models.py#L599)

**Comment by TNA-Allan** [comment]
> * Changing `catalogue_source` to `source`

The data extracted is from `source.value` which negates the earlier notion that its value is exclusive for Catalogue/CAT. Therefore with this change of name it would accommodate - same name as api, and the purpose of distinguishing  Catalogue records from Archive records.

**Comment by TNA-Allan** [comment]
> * Front-end changes (scss and template contents)

These changes are brought it from another branch to be integrated with the API response. No PR for this, as it was discussed to do integrate it with the API response. This is the merge commit of [integrating HTML,CSS in this PR](https://github.com/nationalarchives/ds-wagtail/pull/752/commits/979ac27d9baaf03bea8548a4a6333ff6eb5678da). 

Thanks @ababic its a good point and will take with any similar upcoming tickets which requires a new page.

File: etna/collections/tests/test_views.py
Comment (TNA-Allan) [reply etna/collections/tests/test_views.py:31]: Same comment as replied for 
> Changing title to summary_title

File: etna/records/models.py
Comment (TNA-Allan) [reply etna/records/models.py:235]: Same comment as replied to

> Changing catalogue_source to source

File: etna/records/models.py
Comment (TNA-Allan) [reply etna/records/models.py:529]: HTML is expected here, and we have to trust the source - api and those creating the HTML, just like for example [return mark_safe(self._get_raw_description())](https://github.com/nationalarchives/ds-wagtail/blob/11349d30a0c2d919bc779131486ee3e3ec522015/etna/records/models.py#L230)

File: etna/records/models.py
Comment (TNA-Allan) [reply etna/records/models.py:532]: HTML is expected here, and we have to trust the source - api and those creating the HTML, just like for example [return mark_safe(self._get_raw_description())](https://github.com/nationalarchives/ds-wagtail/blob/11349d30a0c2d919bc779131486ee3e3ec522015/etna/records/models.py#L230)

File: etna/ciim/constants.py
Comment (ababic) [reply etna/ciim/constants.py:721]: Yes, but the difference is: `Bucket` and `BucketList` are USED only in constants. Not in other modules.

Comment (TNA-Allan) [reply etna/ciim/constants.py]: Relocated the new dataclass to etna\records\classes.py since the data is transformed and not used only in constants.

File: etna/records/models.py
Comment (TNA-Allan) [reply etna/records/models.py]: Have conditionally set url, when that data is available and there is a match.

File: etna/records/models.py
Comment (TNA-Allan) [reply etna/records/models.py]: 1. Have used return None when there is no data available to extract.
2. Updated docstrings.

File: etna/records/models.py
Comment (TNA-Allan) [reply etna/records/models.py]: 1. Have used return None when there is no data available to extract.
2. Updated docstrings.

File: etna/ciim/tests/factories.py
Comment (ababic) [inline etna/ciim/tests/factories.py:16]: How do you feel about `source_value_overrides` as a name, and adding a type hint here to indicate that an `Optional[Dict[str, Any]]` is expected?

**Comment by ababic** [comment]
Hi @TNA-Allan. The only outstanding thing here not is the unit tests for the new `Record` class methods:

> Separately, I would say that these new Record methods definitely require unit tests. Maybe with so much going on in each method, it might actually be good to move the "XML to dataclass instance conversion" logic to the dataclasses themselves (e.g. by adding a `from_xml_string()` classmethod)

NOTE: If you did move the conversion logic to the dataclasses, I think you'd have an easier time writing unit tests.

**Comment by TNA-Allan** [comment]
> Hi @TNA-Allan. The only outstanding thing here is the unit tests for the new `Record` class methods:
> 
> > Separately, I would say that these new Record methods definitely require unit tests. Maybe with so much going on in each method, it might actually be good to move the "XML to dataclass instance conversion" logic to the dataclasses themselves (e.g. by adding a `from_xml_string()` classmethod)
> 
> NOTE: If you did move the conversion logic to the dataclasses, I think you'd have an easier time writing unit tests.

The tests for the new `Record` class methods are available in etna/records/tests/test_models.py - its collapsed as the diff is large.  As such these tests are integrated with the API in the tests than to test the new dataclass separately which is without the API integration. Thoughts?

**Comment by ababic** [comment]
> The tests for the new Record class methods are available in etna/records/tests/test_models.py - its collapsed as the diff is large. As such these tests are integrated with the API in the tests than to test the new dataclass separately which is without the API integration. Thoughts?

Okay, thanks for that. I did miss those. I'll leave that up to you then. Even if you refactored, you would still want some integration tests. For the most part, you would just be throwing strings at method and checking the output (so, no API mocking etc needed).

**Review by ababic — APPROVED** [review]
Approving with optional suggestions on the XML conversion refactoring (You could add a 'Tech debt ticket' to the Jira backlog for this, and tackle it later)

**Comment by TNA-Allan** [comment]
> Approving with optional suggestions on the XML conversion refactoring (You could add a 'Tech debt ticket' to the Jira backlog for this, and tackle it later)

Sure, Thanks Andy.

**Pull request merged**
